pkg/did/client.go

```diff
@@ -115,34 +115,30 @@ func (c *Client) getEndpoints(domain string) ([]*endpoint.Endpoint, error) {
 
 // buildSideTreeRequest request builder for sidetree public DID creation
 func (c *Client) buildSideTreeRequest(createDIDOpts *CreateDIDOpts) ([]byte, error) {
-	didDoc := createDIDOpts.didDoc
-
-	// create default did doc if user didn't provide their DID
-	if didDoc == nil {
-		publicKeys := createDIDOpts.publicKeys
-
-		// create default public key if user didn't provide their public key
-		if len(publicKeys) == 0 {
-			_, base58PubKey, err := c.kms.CreateKeySet()
-			if err != nil {
-				return nil, fmt.Errorf("failed to create key set: %w", err)
-			}
-
-			publicKeys = append(publicKeys, docdid.PublicKey{
-				ID:    pubKeyIndex1,
-				Type:  defaultKeyType,
-				Value: base58.Decode(base58PubKey),
-			})
-		}
+	publicKeys := createDIDOpts.publicKeys
 
-		didDoc = &docdid.Doc{
-			Context:   []string{},
-			PublicKey: publicKeys,
-			Service:   createDIDOpts.services,
+	// create default public key if user didn't provide their public key
+	if len(publicKeys) == 0 {
+		_, base58PubKey, err := c.kms.CreateKeySet()
+		if err != nil {
+			return nil, fmt.Errorf("failed to create key set: %w", err)
 		}
+
+		publicKeys = append(publicKeys, PublicKey{
+			ID:       pubKeyIndex1,
+			Type:     defaultKeyType,
+			Encoding: PublicKeyEncodingBase58,
+			Value:    base58.Decode(base58PubKey),
```

Comment:
Is this right? You specify the encoding as base58 and then you decode it?

Reply:
yes the Encoding flag doesn't mean the value in base58

Reply:
it just way to tell encoder to encode this value to base58 https://github.com/trustbloc/trustbloc-did-method/pull/89/files/14b9149e0ae51efb373187cf544ac04f6b68d057#diff-4d8c1b4ae48f117488270dedd9fb45caR97